x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_provider.tsx

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect } from 'react';
+import { useValues, useActions } from 'kea';
+import { History } from 'history';
+
+import {
+  FlashMessagesLogic,
+  IFlashMessagesLogicValues,
+  IFlashMessagesLogicActions,
+} from './flash_messages_logic';
+
+interface IFlashMessagesProviderProps {
+  history: History;
+}
+
+export const FlashMessagesProvider: React.FC<IFlashMessagesProviderProps> = ({ history }) => {
+  const { historyListener } = useValues(FlashMessagesLogic) as IFlashMessagesLogicValues;
+  const { listenToHistory } = useActions(FlashMessagesLogic) as IFlashMessagesLogicActions;
+
+  useEffect(() => {
+    if (!historyListener) listenToHistory(history);
+  }, []);
```

Comment:
It's a best practice to never leave empty arrays in useEffect calls. If and when this is added to the linter, I believe it would require `[historyListener, history]` to pass

Reply:
[Reference](https://medium.com/@5066aman/effects-shouldnt-lie-about-their-dependencies-44becff15582)

Reply:
We're not linting for it. We turned off that rule for Enterprise Search since the MVP in `.eslintrc.js` here:

https://github.com/constancecchen/kibana/pull/16/commits/fa5ecdaf93ca72715b0ea95d7d0d62c03e5beb52

As to whether it's a best practice - quite frankly, I'm not convinced it is in all cases and [there are a not insignificant amount of people who agree](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485963251).

For example, in the case where I had to disable the rule for App Search's Engine Overview:

https://github.com/constancecchen/kibana/blob/fa5ecdaf93ca72715b0ea95d7d0d62c03e5beb52/x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx#L74-L75

It wanted me to add `setEngineData` to the dependencies list which was simply a reusable helper function I'd defined further up in the page. However, adding it to the dependency list causes the component to constantly refresh/make async calls since `const setEngineData` is defined on every component update. It's a bizarre circular loop that frankly doesn't need to exist, since it's just a helper func and not actually something I want the page to update on if it changes.

That's not the case in this particular scenario, so I don't have any issues making the change - I'm just not really convinced how valuable it is here. Just my personal preference, but I like that an empty array has a very specific meaning: this runs on mount, and that's it. It's a paradigm those familiar with the old `componentDidMount` cycle are used to, and for providers like this that handle setting up a store on plugin init, it makes sense (HttpLogic also does this, FWIW).

Reply:
I agree with you there Constance 👍